src/main/mondrian/spi/impl/SparkSqlDialect.java

```diff
@@ -0,0 +1,176 @@
+/*
+// This software is subject to the terms of the Eclipse Public License v1.0
+// Agreement, available at the following URL:
+// http://www.eclipse.org/legal/epl-v10.html.
+// You must accept the terms of that agreement to use this software.
+//
+// Copyright (C) 2011-2014 Pentaho and others
+// All Rights Reserved.
+*/
+package mondrian.spi.impl;
+
+import mondrian.olap.Util;
+
+import java.sql.*;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Implementation of {@link mondrian.spi.Dialect} for the SparkSql database.
+ *
+ * @author Hongwei Fu
+ * @since Jan 10, 2011
+ */
+public class SparkSqlDialect extends JdbcDialectImpl {
+    private static final int MAX_COLUMN_NAME_LENGTH = 128;
+
+    public static final JdbcDialectFactory FACTORY =
+        new JdbcDialectFactory(
+            SparkSqlDialect.class,
+            DatabaseProduct.SPARKSQL)
+        {
+            protected boolean acceptsConnection(Connection connection) {
+                return super.acceptsConnection(connection);
+            }
+        };
+
+    /**
+     * Creates a SparkSqlDialect.
+     *
+     * @param connection Connection
+     *
+     * @throws java.sql.SQLException on error
+     */
+    public SparkSqlDialect(Connection connection) throws SQLException {
+        super(connection);
+    }
+
+    protected String deduceIdentifierQuoteString(
```

Comment:
No quotes at all? This won't work with drillthrough operations, where the column alias might be two words. Not quoting them will create invalid SQL.